File: quan_ly_cong_viec_app/lib/models/task.dart
class Task {
  final int id;
  final String title;
  final String? note;
  final String dueDate;
  final String? dueTime; // <-- THÊM TRƯỜNG MỚI
  bool isCompleted;

  Task({
    required this.id,
    required this.title,
    this.note,
    required this.dueDate,
    this.dueTime, // <-- CẬP NHẬT CONSTRUCTOR
    required this.isCompleted,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      note: json['note'],
      dueDate: json['due_date'],
      dueTime: json['due_time'], // <-- LẤY DỮ LIỆU TỪ JSON
      // API trả về 0 hoặc 1 cho boolean, cần chuyển đổi
      isCompleted: json['is_completed'] is bool
          ? json['is_completed']
          : json['is_completed'] == 1,
    );
  }
}

File: quan_ly_cong_viec_api/app/Models/Task.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Task extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'title',
        'note',
        'due_date',
        'due_time', // <-- ĐÃ THÊM
        'is_completed',
        'is_fixed',
        'type',
        'user_id',
    ];

    /**
     * The attributes that should be cast.
     * Giúp Laravel tự động chuyển đổi kiểu dữ liệu.
     * @var array<string, string>
     */
    protected $casts = [
        'is_completed' => 'boolean',
        'is_fixed' => 'boolean',
    ];

    /**
     * Lấy user sở hữu công việc này.
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

